core/src/main/java/gyro/core/reference/FinderReferenceResolver.java

```diff
@@ -46,14 +49,16 @@ public Object resolve(Scope scope, List<Object> arguments) {
                 type));
         }
 
-        Finder<Resource> finder = FinderType.getInstance(finderClass).newInstance(scope);
+        FinderType<? extends Finder<Resource>> finderType = FinderType.getInstance(finderClass);
+        Finder<Resource> finder = finderType.newInstance(scope);
```

Comment:
Could these lines move to before line 61 since that's the only part of the code that uses them?

Reply:
52 is used by 53 and 53 is used by 67, so both of them cannot be moved